DiscordChatExporter.Core/Exporting/Filtering/Parsing/FilterTokenizer.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Superpower;
+using Superpower.Parsers;
+using Superpower.Tokenizers;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering.Parsing
+{
+    public static class FilterTokenizer
```

Comment:
What do you think, can we get away without having a tokenizer and just work with raw characters directly? I don't really care about error messages _too much_, but I wonder if writing a scanless parser would reduce complexity a bit.

Reply:
In terms of error messages I think the only real benefit of the tokenizer is being able to identify "string" since the rest are single characters anyway, and we could probably replicate that message elsewhere. I think being able to handle all whitespace in the tokenizer does reduce the parser complexity though, since that keeps us from having to worry about it in a bunch of other places. We probably wouldn't want to be rejecting things like `from: foo` vs. `from:foo`, and it might be messy to handle various cases like that in multiple parsers.

Reply:
Good point